### ISSUE TITLE
Solc error log shown when application throws an error

### ISSUE THREAD

**Issue by miguelmtzinf** [ethereum/solc-js#505]
When my application crashes (exception thrown with no handler) the library solc (using latest version) show an error which should not be shown (since It is not an error from solc itself)

```
/home/myuser/solc-spike/node_modules/solc/soljson.js:32
(Module.asmGlobalArg,Module.asmLibraryArg,buffer);var __GLOBAL__I_000101=Module["__GLOBAL__I_000101"]=asm["__GLOBAL__I_000101"];var __GLOBAL__sub_I_ABIFunctions_cpp=Module["__GLOBAL__sub_I_ABIFunctions_cpp"]=asm["__GLOBAL__sub_I_ABIFunctions_cpp"];var __GLOBAL__sub_I_ABI_cpp=Module["__GLOBAL__sub_I_ABI_cpp"]=asm["__GLOBAL__sub_I_ABI_cpp"];var __GLOBAL__sub_I_ASTJsonConverter_cpp=Module["__GLOBAL__sub_I_ASTJsonConverter_cpp"]=asm["__GLOBAL__sub_I_ASTJsonConverter_cpp"];var __GLOBAL__sub_I_AST_cpp=Module["__GLOBAL__sub_I_AST_cpp"]=asm["__GLOBAL__sub_I_AST_cpp"];var __GLOBAL__sub_I_ArrayUtils_cpp=Module["__GLOBAL__sub_I_ArrayUtils_cpp"]=asm["__GLOBAL__sub_I_ArrayUtils_cpp"];var __GLOBAL__sub_I_AsmAnalysis_cpp=Module["__GLOBAL__sub_I_AsmAnalysis_cpp"]=asm["__GLOBAL__sub_I_AsmAnalysis_cpp"];var __GLOBAL__sub_I_BMC_cpp=Module["__GLOBAL__sub_I_BMC_cpp"]=asm["__GLOBAL__sub_I_BMC_cpp"];var __GLOBAL__sub_I_CHC_cpp=Module["__GLOBAL__s
abort({"errno":-21,"code":"EISDIR","syscall":"read"}). Build with -s ASSERTIONS=1 for more info.
```

(I think related to ESM encryption, I see old issues about this, but still happening...)

Code snippet:

```


const solc = require('solc');


(async () => {
  throw new Error('bla');
})()
//.catch((err) => {
//  console.log('ei')
//});
```

Running this code snippet will show an error from solc with no reason.... If you uncomment last 3 lines, you will remove the error. Why solc is handling uncaught exceptions?

**Comment by Leask** [comment]
This may help:

`// solc init
const solc = require('solc');
const uncRjc = 'unhandledRejection';
const listeners = process.listeners(uncRjc);
process.removeListener(uncRjc, listeners[listeners.length - 1]);
`

**Comment by cameel** [comment]
There was a duplicate of this issue reported and I moved it to the solidity repo because it's really a problem with the binary, not solc-js. I'm not closing this one to make it visible to anyone else who would potentially want to report any discussion should go into https://github.com/ethereum/solidity/issues/12228 instead.

**Comment by nikola-matic** [comment]
Should be solved as of `0.8.12` (see https://github.com/ethereum/solidity/issues/12228#issuecomment-1293993189)
